### PR TITLE
Assign pull request to project during creation

### PR DESCRIPTION
**Pull request by denyskon**
When creating a pull request, allow directly assigning it to a project, as it is already possible for issues.

After:

grafik

### CODE REVIEW

**Comment by lunny** [comment]
I think only those who have write permission could change it.

**Comment by denyskon** [comment]
@lunny This is being checked inside the template. It's the same logic as with issues

**Comment by delvh** [comment]
I would even go further as to unify the templates.
I've never understood the reason for offering different options.

**Comment by denyskon** [comment]
@delvh It uses the same template, just the projects variable wasn't set in backend for pull requests.

**Comment by delvh** [comment]
I meant during PR creation as well when viewing a created PR.
I don't see why we should differentiate between the two.

**Comment by denyskon** [comment]
We could move that sidebar out to a subtemplate. There are things we need to hide like time tracking, but otherwise it should be possible. This sounds like a bigger refactoring however, IMO out of scope for this PR

**Review by delvh — APPROVED** [review]

File: routers/web/repo/pull.go [hunk 1]
```diff
@@ -1457,6 +1457,18 @@ func CompareAndPullRequestPost(ctx *context.Context) {
 		return
 	}
 
+	if projectID > 0 {
+		if !ctx.Repo.CanRead(unit.TypeProjects) {
```
Comment (delvh) [inline routers/web/repo/pull.go]: Is `read projects` the correct permission for this?

Comment (denyskon) [reply routers/web/repo/pull.go]: I just copied this logic from the issue router.... I think it isn't problematic as the write check already happens in the template before offering the "set project" option

Comment (lunny) [reply routers/web/repo/pull.go]: It's not right. It should be write permission. Issue permission is special.

Comment (denyskon) [reply routers/web/repo/pull.go]: Okay, I changed it to write

**Comment by delvh** [comment]
ping for other reviewers

**Review by lunny — APPROVED** [review]

**Pull request merged**